managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -429,6 +429,9 @@ public void operationComplete(ManagedLedgerInfo mlInfo, Stat stat) {
                                 log.debug("[{}] Opened ledger {}: {}", name, id, BKException.getMessage(rc));
                             }
                             if (rc == BKException.Code.OK) {
+                                if (State.Terminated.equals(state)) {
```

Comment:
```
State state = STATE_UPDATER.get(this);
state == State.Terminated
```

Reply:
I don't think that `STATE_UPDATER.get` is needed. It makes no difference when reading volatile variables.